src/runtime_src/core/tools/xbflash2/SubCmdProgram.cpp

```diff
@@ -0,0 +1,136 @@
+    : SubCmd("program", 
+             "Updates the image(s) for a given device")
+{
+  const std::string longDescription = "Programs the given acceleration image into the device's shell.";
+  setLongDescription(longDescription);
+  setExampleSyntax("");
+  setIsHidden(_isHidden);
+  setIsDeprecated(_isDepricated);
+  setIsPreliminary(_isPreliminary);
+}
+
+
+void
+SubCmdProgram::execute(const SubCmdOptions& _options) const
+{
+  // -- Common top level options ---
+  bool help = false;
+
+  po::options_description commonOptions("Common Options"); 
+  commonOptions.add_options()
+    ("help", boost::program_options::bool_switch(&help), "Help to use this sub-command")
+  ;
+
+  po::options_description hiddenOptions("Hidden Options"); 
+
+  // -- Define the supporting option options ----
+  SubOptionOptions subOptionOptions;
+  subOptionOptions.emplace_back(std::make_shared<OO_Program_Spi>("spi"));
+  subOptionOptions.emplace_back(std::make_shared<OO_Program_Qspips>("qspips"));
+
+  for (auto & subOO : subOptionOptions) {
+    if (subOO->isHidden()) 
+      hiddenOptions.add_options()(subOO->longName().c_str(), subOO->description().c_str());
+    else
+      commonOptions.add_options()(subOO->longName().c_str(), subOO->description().c_str());
+    subOO->setExecutable(getExecutableName());
+    subOO->setCommand(getName());
+  }
+
+  po::options_description allOptions("All Options");
+  allOptions.add(commonOptions);
+  allOptions.add(hiddenOptions);
+
+  // =========== Process the options ========================================
+
+  // 1) Process the common top level options 
+  po::parsed_options parsedCommonTop = 
+    po::command_line_parser(_options).
+    options(allOptions).          
+    allow_unregistered().           // Allow for unregistered options
+    run();                          // Parse the options
+
+  po::variables_map vm;
+
+  try {
+    po::store(parsedCommonTop, vm);  // Can throw
+    po::notify(vm);                  // Can throw (but really isn't used)
+
+    // Mutual DRC
+    for (unsigned int index1 = 0; index1 < subOptionOptions.size(); ++index1) {
+      for (unsigned int index2 = index1 + 1; index2 < subOptionOptions.size(); ++index2) {
+        conflictingOptions(vm, subOptionOptions[index1]->longName(), subOptionOptions[index2]->longName());
+      }
+    }
+  } catch (const std::exception & e) {
+    std::cerr << "ERROR: " << e.what() << std::endl;
+    printHelp(commonOptions, hiddenOptions, subOptionOptions);
+    throw std::errc::operation_canceled;
+  }
+
+  // Find the subOption;
+  std::shared_ptr<OptionOptions> optionOption;
+  for (auto& subOO : subOptionOptions) {
+    if (vm.count(subOO->longName().c_str()) != 0) {
+      optionOption = subOO;
+      break;
+    }
+  }
+
+  // No suboption print help
+  if (!optionOption) {
+      if (help) {
+          printHelp(commonOptions, hiddenOptions, subOptionOptions);
+          return;
+      }
+      else {
```

Comment:
No need for an else for the above if block returns from this function.
```suggestion
      else {
```